pkg/drift/drift.go

```diff
@@ -0,0 +1,137 @@
+	}
+	buckets, err := assetsClient.GetBuckets(ctx, organizationID, bucketQuery)
+	if err != nil {
+		return fmt.Errorf("error fetching terraform state GCS buckets: %w", err)
+	}
+	fmt.Printf("Fetching IAM for %d Folders and %d Projects\n", len(folders), len(projects))
+
+	gcpIAM, err := getActualGCPIAM(ctx, organizationID, folders, projects)
+	if err != nil {
+		return fmt.Errorf("error determining GCP IAM: %w", err)
+	}
+	fmt.Printf("Fetching terraform state from %d Buckets\n", len(buckets))
+	tfIAM, err := getTerraformStateIAM(ctx, organizationID, folders, projects, buckets)
+	if err != nil {
+		return fmt.Errorf("error determining Terraform State: %w", err)
+	}
+	fmt.Printf("Found %d gcp IAM entries\n", len(gcpIAM))
+	fmt.Printf("Found %d terraform IAM entries\n", len(tfIAM))
+
+	clickOpsChanges := difference(gcpIAM, tfIAM)
+	missingTerraformChanges := difference(tfIAM, gcpIAM)
+
+	fmt.Printf("Found %d Click Ops Changes\n", len(clickOpsChanges))
+	fmt.Printf("Found %d Missing Terraform Changes\n", len(missingTerraformChanges))
+
+	return nil
+}
+
+func getActualGCPIAM(
+	ctx context.Context,
+	organizationID int64,
+	folders []assets.HierarchyNode,
+	projects []assets.HierarchyNode,
+) (map[string]iam.AssetIAM, error) {
+	client, err := iam.NewClient(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("error initializing iam Client: %w", err)
+	}
+
+	m := make(map[string]iam.AssetIAM)
+	for _, f := range folders {
+		fIAM, err := client.GetIAMForFolder(ctx, f, "folders")
+		if err != nil {
+			return nil, fmt.Errorf("unable to get folder IAM for folder with ID '%d' and name '%s': %w", f.ID, f.Name, err)
+		}
+		for _, i := range fIAM {
+			m[iam.CreateURI(i, organizationID)] = i
+		}
+	}
+	for _, p := range projects {
+		pIAM, err := client.GetIAMForProject(ctx, p, "projects")
+		if err != nil {
+			return nil, fmt.Errorf("unable to get project IAM for project with ID '%d' and name '%s': %w", p.ID, p.Name, err)
+		}
+		for _, i := range pIAM {
+			m[iam.CreateURI(i, organizationID)] = i
+		}
+	}
+
+	return m, nil
+}
+
+func getTerraformStateIAM(
+	ctx context.Context,
+	organizationID int64,
+	folders []assets.HierarchyNode,
+	projects []assets.HierarchyNode,
+	gcsBuckets []string,
+) (map[string]iam.AssetIAM, error) {
+	parser, err := terraform.NewParser(ctx, organizationID, folders, projects)
+	if err != nil {
+		return nil, fmt.Errorf("error initializing terraform parser: %w", err)
+	}
+	gcsURIs, err := parser.GetAllTerraformStateFileURIs(ctx, gcsBuckets)
+	if err != nil {
+		return nil, fmt.Errorf("error determining terraform state files: %w", err)
+	}
+
+	tIAM, err := parser.ProcessAllTerraformStates(ctx, gcsURIs)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse terraform states: %w", err)
+	}
+
+	m := make(map[string]iam.AssetIAM)
+	for _, i := range tIAM {
+		m[iam.CreateURI(i, organizationID)] = i
+	}
+
+	return m, nil
+}
+
+func difference(source, target map[string]iam.AssetIAM) []string {
```

Comment:
This only finds things in `source` that are not in target; it does not find things in `target` that are not in `source`.